rlberry/utils/check_agent.py

```diff
@@ -0,0 +1,202 @@
+    one time with 20 fit budget.
+
+    Parameters
+    ----------
+
+    Agent: rlberry agent module
+        Agent class that we want to test.
+
+    continuous_state: bool, default=False
+        whether we test on a discrete state environment or a continuous one.
+    """
+    if continuous_state:
+        env_ctor = PBall2D
+    else:
+        env_ctor = Chain
+    env_kwargs = {}
+    agent1 = AgentManager(
+        Agent, (env_ctor, env_kwargs), fit_budget=5, n_fit=1, seed=SEED
+    )
+    agent1.fit(10)
+    agent1.fit(10)
+
+    agent2 = AgentManager(
+        Agent, (env_ctor, env_kwargs), fit_budget=5, n_fit=1, seed=SEED
+    )
+    agent2.fit(20)
+
+    env = env_ctor()
+    state = env.reset()
+    result = check_agent_almost_equal(state, agent1, agent2)
+
+    assert (
+        result
+    ), "Error: fitting the agent two times for 10 steps is not equivalent to fitting it one time for 20 steps."
+
+
+def check_save(Agent, continuous_state=False):
+    """
+    Check that the agent save a non-empty file.
+
+
+    Parameters
+    ----------
+
+    Agent: rlberry agent module
+        Agent class that we want to test.
+
+    continuous_state: bool, default=False
+        whether we test on a discrete state environment or a continuous one.
+    """
+    if continuous_state:
+        env_ctor = PBall2D
+    else:
+        env_ctor = Chain
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        if continuous_state:
+            env_ctor = PBall2D
+        else:
+            env_ctor = Chain
+        env_kwargs = {}
+        agent1 = AgentManager(
+            Agent,
+            (env_ctor, env_kwargs),
+            fit_budget=5,
+            n_fit=1,
+            seed=SEED,
+            output_dir=tmpdirname,
+        )
+        agent1.fit(10)
+        assert (
+            os.path.getsize(str(agent1.output_dir_) + "/agent_handlers/idx_0.pickle")
+            > 1
+        ), "the file saved appear to be empty"
+
+
+def check_seeding_agent(Agent, continuous_state=False):
+    """
+    Check that the agent is reproducible.
+
+    Parameters
+    ----------
+
+    Agent: rlberry agent module
+        Agent class that we want to test.
+
+    continuous_state: bool, default=False
+        whether we test on a discrete state environment or a continuous one.
+    """
+    agent1 = check_agent_manager(Agent, continuous_state)
+    agent2 = check_agent_manager(Agent, continuous_state)
+
+    if continuous_state:
+        env_ctor = PBall2D
+    else:
+        env_ctor = Chain
+    env = env_ctor()
+    state = env.reset()
+    result = check_agent_almost_equal(state, agent1, agent2)
+
+    assert result, "Agent not reproducible (different seed give different results)"
+
+
+def check_rl_agent(Agent, continuous_state=False):
+    """
+    Check agent manager compatibility  and check reproducibility/seeding.
+    Will raise an exception if there is something wrong with the agent.
```

Comment:
There are already exception in each individual check and I don't want to wrap all of this in a `check_rl_agent` exception because this would hide the real error/exception.

Reply:
Oh, I meant this just as a rewrite of the "Will raise an exception if there is something wrong with the agent." part of the docstring since it seemed long. When I feel like a phrase in a docstring feels long or colloquial, I usually just google docstrings from other libraries (e.g. numpy, scikit) and copy their style. This is what I've done here!